### PR TITLE
build(docker): Create Docker Compose file for local setup WITHOUT using Postgres/LogTo [DEV-2681]

### PR DESCRIPTION
**Pull request by abdulla-ashurov**
- [X] Create docker compose for local setup without using Postgres/LogTo
  - [X] Create credential service in docker compose file
  - [X] Create Postgres DB service in docker compose file
  - [X] Create documentation about how to local up credential-service with docker compose.
- [X] Credential Service: Fix bug with an incorrect logic of ENABLE_EXTERNAL_DB toggle [DEV-2834]

### CODE REVIEW

**Review by DaevMithran — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -50,34 +50,39 @@ The application allows configuring the following parameters using environment va
...
 
-1. `DB_CONNECTION_URL`: Postgres database connection URL, e.g. `postgres://<user>:<password>@<host>:<port>/<database>`
-2. `DB_ENCRYPTION_KEY`: Secret key used to encrypt the Veramo key-specific database tables. This adds a layer of protection by not storing the database in plaintext.
-3. `DB_CERTIFICATE`: Custom CA certificate required to connect to the database (optional).
+1. `ENABLE_EXTERNAL_DB`: Turns external database on/off (Default: `false`). If `ENABLE_EXTERNAL_DB=true`, then define below environment variables in `.env` file:
+    - `DB_CONNECTION_URL`: Postgres database connection URL, e.g. `postgres://<user>:<password>@<host>:<port>/<database>`.
+    - `DB_ENCRYPTION_KEY`: Secret key used to encrypt the Veramo key-specific database tables. This adds a layer of protection by not storing the database in plaintext.
+    - `DB_CERTIFICATE`: Custom CA certificate required to connect to the database (optional).
+    - `POSTGRES_USER`: Postgres database username using in database connection URL.
```
Comment (anikitinDSR) [inline README.md:67]: Let's create another section, specially for local setup, just not to mess with staging/production environment

File: docker/.env.example [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+RESOLVER_URL="https://resolver.cheqd.net/1.0/identifiers/"
+APPLICATION_BASE_URL="http://localhost:8787"
+ALLOWED_ORIGINS="http://localhost:8787"
+
+# Database
+ENABLE_EXTERNAL_DB="string,default:false"
+EXTERNAL_DB_CONNECTION_URL="postgres://<user>:<password>@<host>:<port>/<database>"
+EXTERNAL_DB_ENCRYPTION_KEY="<db-encryption-key>"
+EXTERNAL_DB_CERT="<path/to/extenal-db-cert>"
+POSTGRES_USER="<user>"
```
Comment (anikitinDSR) [inline docker/.env.example:12]: Is it possible to pass into docker container DB_URL instead of several vars with username/password?

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    # Default is to pull in the pre-published image on GitHub Container Registry
+    # SYNTAX: docker compose -f docker/docker-compose.yml build
+    # build:
+    #   context: ../
+    #   dockerfile: docker/Dockerfile
+    # CAUTION: Change image section's value if building your own image in section below
+    image: ghcr.io/cheqd/credential-service:staging-latest
+    ports:
+      - 8787:8787
+    env_file:
```
Comment (anikitinDSR) [inline docker/docker-compose.yml:22]: this one is not needed cause it by default searches it

Comment (abdulla-ashurov) [reply docker/docker-compose.yml:22]: Seems, it doesn't search it. Instead of that I need to define every env variables in docker-compose.yml: https://docs.docker.com/compose/environment-variables/set-environment-variables/

File: README.md (hunk 1 above)
Comment (abdulla-ashurov) [reply README.md]: Done! ✅

File: docker/.env.example (hunk 2 above)
Comment (abdulla-ashurov) [reply docker/.env.example:12]: No, It isn't.

File: docker/Dockerfile [hunk 4]
```diff
@@ -42,17 +45,18 @@ RUN npm ci
 
 # Base arguments: build-time
 ARG NPM_CONFIG_LOGLEVEL=warn
-ARG PORT=3000
+ARG PORT=8787
```
Comment (anikitinDSR) [inline docker/Dockerfile:48]: why did you change this port?
I think it can potential affect our staging/production deploy.

File: docker/Dockerfile [hunk 5]
```diff
@@ -42,17 +45,18 @@ RUN npm ci
...
 # Network API endpoints: build-time
 ARG MAINNET_RPC_URL=https://rpc.cheqd.net:443
 ARG TESTNET_RPC_URL=https://rpc.cheqd.network:443
 ARG RESOLVER_URL=https://resolver.cheqd.net/1.0/identifiers/
 
 # Veramo Database configuration: build-time
-ARG DB_CONNECTION_URL
-ARG DB_ENCRYPTION_KEY
-ARG DB_CERTIFICATE
+ARG ENABLE_EXTERNAL_DB
```
Comment (anikitinDSR) [inline docker/Dockerfile:56]: could we also specify a default value for this variable?

File: docker/Dockerfile (hunk 4 above)
Comment (abdulla-ashurov) [reply docker/Dockerfile]: Done! ✅

File: docker/Dockerfile (hunk 5 above)
Comment (abdulla-ashurov) [reply docker/Dockerfile]: Done! ✅
